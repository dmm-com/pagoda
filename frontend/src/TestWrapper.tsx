import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
