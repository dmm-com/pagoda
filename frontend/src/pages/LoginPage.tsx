import ExitToApp from "@mui/icons-material/ExitToApp";
import InfoIcon from "@mui/icons-material/Info";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import {
  Alert,
  Box,
  Button,
  Checkbox,
  FormControl,
  FormControlLabel,
  FormGroup,
  FormHelperText,
  IconButton,
  Input,
  InputAdornment,
  Link,
  TextField,
  Typography,
} from "@mui/material";
import React, { FC, useCallback, useEffect, useState } from "react";
import { useHistory } from "react-router-dom";

import { PasswordResetConfirmModal } from "../components/user/PasswordResetConfirmModal";
import { PasswordResetModal } from "../components/user/PasswordResetModal";
import { aironeApiClient } from "../repository/AironeApiClient";

import { ServerContext } from "services/ServerContext";

export const LoginPage: FC = () => {
  const serverContext = ServerContext.getInstance();
  const history = useHistory();

  const [showPassword, setShowPassword] = useState(false);
  const [isAlert, setIsAlert] = useState(false);
  const [showAlertForTermsOfService, setShowAlertForTermsOfService] =
    useState(false);
  const [agreeWithServiceContract, setAgreeWithServiceContract] =
    useState(false);
  const [openPasswordResetModal, setOpenPasswordResetModal] = useState(false);
  const [uidb64, setUidb64] = useState<string>("");
  const [token, setToken] = useState<string>("");
  const [openPasswordResetConfirmModal, setOpenPasswordResetConfirmModal] =
    useState(false);

  useEffect(() => {
    const params = new URLSearchParams(location.search);

    const _uidb64 = params.get("uidb64");
    if (_uidb64 != null) {
      setUidb64(_uidb64);
    }

    // get token from query parameter, then delete it to avoid leaking via Referer header(a similar logic to Django auth module).
    const _token = params.get("token");
    if (_token != null) {
      setToken(_token);
      params.delete("token");
      history.replace({
        pathname: location.pathname,
        search: "?" + params.toString(),
      });
    }

    if (_uidb64 != null && _token != null) {
      setOpenPasswordResetConfirmModal(true);
    }
  }, [location.search]);

  const handleClickShowPassword = () => {
    setShowPassword(!showPassword);
  };

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setIsAlert(false);

    const data = new FormData(event.currentTarget);

    // set AGREE_TERM_OF_SERVICE when user indicates to agree with Pagoda's service contract
    // when serverContext.checkTermService is true.
    if (serverContext?.checkTermService) {
      if (agreeWithServiceContract === true) {
        data.append(
          "extra_param",
          JSON.stringify({ AGREE_TERM_OF_SERVICE: true })
        );
      } else {
        // abort login process when user does not agree with Pagoda's service contract
        setShowAlertForTermsOfService(true);
        return;
      }
    }

    const resp = await aironeApiClient.postLogin(data);
    if (resp.type === "opaqueredirect") {
      window.location.href = serverContext?.loginNext ?? "";
    } else {
      setIsAlert(true);
    }
  };

  const handleOpenPasswordResetModal = useCallback(() => {
    setOpenPasswordResetModal(true);
  }, [setOpenPasswordResetModal]);

  const handleClosePasswordResetModal = useCallback(() => {
    setOpenPasswordResetModal(false);
  }, [setOpenPasswordResetModal]);

  const handleClosePasswordResetConfirmModal = useCallback(() => {
    setOpenPasswordResetConfirmModal(false);
  }, [setOpenPasswordResetConfirmModal]);

  console.log("[onix(50)] serverContext: ", serverContext);
  return (
    <Box
      display="flex"
      justifyContent="center"
      alignItems="center"
      bgcolor="primary.main"
      height="100vh"
    >
      <Box
        width="60%"
        minWidth="600px"
        height="70%"
        minHeight="500px"
        display="flex"
        flexDirection="column"
        justifyContent="center"
        alignItems="center"
        bgcolor="white"
      >
        <Typography variant="h1" fontWeight={400}>
          {serverContext?.title}
        </Typography>
        <Typography variant="subtitle2" mt={2}>
          {serverContext?.subTitle}
        </Typography>
        {
          serverContext?.checkTermService && (
            <Box width={500}>
              <FormControl>
                <FormGroup>
                  <FormControlLabel
                    control={
                      <Checkbox
                        onChange={(e) => {
                          setAgreeWithServiceContract(e.target.checked);
                          setShowAlertForTermsOfService(false);
                        }}
                      />
                    }
                    label="以下の規約に合意する。"
                  />
                  <Link href={serverContext?.termsOfServiceUrl}>Pagoda サービス規約</Link>
                </FormGroup>
                <FormHelperText>
                  {showAlertForTermsOfService && (
                    <Alert
                      severity="error"
                      onClose={() => {
                        setShowAlertForTermsOfService(false);
                      }}
                    >
                      ご利用には、サービス規約への合意が必要です。
                    </Alert>
                  )}
                </FormHelperText>
              </FormControl>
            </Box>
          )
        }
        <Box width={500} height={50} mt={2}>
          {isAlert ? (
            <Alert
              severity="error"
              onClose={() => {
                setIsAlert(false);
              }}
            >
              ユーザ名またはパスワードが違います。
            </Alert>
          ) : (
            ""
          )}
        </Box>
        <Box
          component="form"
          onSubmit={handleSubmit}
          width={500}
          display="flex"
          flexDirection="column"
          alignItems="center"
        >
          <TextField
            id="username"
            name="username"
            label="Username"
            fullWidth={true}
            margin="dense"
          />
          <TextField
            id="password"
            name="password"
            label="Password"
            type={showPassword ? "text" : "password"}
            fullWidth={true}
            margin="dense"
            InputProps={{
              endAdornment: (
                <InputAdornment position="end">
                  <IconButton onClick={handleClickShowPassword} edge="end">
                    {showPassword ? <VisibilityOff /> : <Visibility />}
                  </IconButton>
                </InputAdornment>
              ),
            }}
          />
          <Input
            id="next"
            name="next"
            type="hidden"
            value={serverContext?.loginNext}
          />
          <Box display="flex" flexDirection="column" width="100%" my="8px">
            {serverContext?.singleSignOnLoginUrl != null && (
              <Link
                color="secondary"
                href={serverContext.singleSignOnLoginUrl}
                sx={{ cursor: "pointer" }}
              >
                <ExitToApp
                  sx={{
                    fontSize: "14px",
                    verticalAlign: "middle",
                  }}
                />
                <Typography fontSize="16px" ml={1} display="inline">
                  SSO ログイン
                </Typography>
              </Link>
            )}
            <Link
              color="secondary"
              sx={{ cursor: "pointer" }}
              href={serverContext?.noteLink}
            >
              <InfoIcon
                sx={{
                  fontSize: "14px",
                  verticalAlign: "middle",
                }}
              />
              <Typography fontSize="16px" ml={1} display="inline">
                {serverContext?.noteDesc}
              </Typography>
            </Link>
            {serverContext?.passwordResetDisabled ? (
              <Typography
                color="secondary"
                fontSize="16px"
                ml={1}
                display="inline"
                sx={{ textDecoration: "line-through" }}
              >
                パスワードリセット
              </Typography>
            ) : (
              <Link
                color="secondary"
                onClick={handleOpenPasswordResetModal}
                sx={{ cursor: "pointer" }}
              >
                <InfoIcon
                  sx={{
                    fontSize: "14px",
                    verticalAlign: "middle",
                  }}
                />
                <Typography fontSize="16px" ml={1} display="inline">
                  パスワードリセット
                </Typography>
              </Link>
            )}
          </Box>
          <Button
            type="submit"
            variant="contained"
            sx={{ mt: 7, width: "auto", py: 0.5, letterSpacing: 1.25 }}
          >
            Login
          </Button>
        </Box>
      </Box>

      <PasswordResetModal
        openModal={openPasswordResetModal}
        closeModal={handleClosePasswordResetModal}
      />
      <PasswordResetConfirmModal
        openModal={openPasswordResetConfirmModal}
        closeModal={handleClosePasswordResetConfirmModal}
        uidb64={uidb64}
        token={token}
      />
    </Box>
  );
};
