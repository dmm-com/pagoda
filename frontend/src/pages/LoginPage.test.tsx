/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";

import { LoginPage } from "./LoginPage";

import { TestWrapper } from "TestWrapper";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  Object.defineProperty(window, "django_context", {
    value: {
      title: "AirOne",
    },
    writable: false,
  });

  const result = render(<LoginPage />, {
    wrapper: TestWrapper,
  });

  expect(result).toMatchSnapshot();
});
