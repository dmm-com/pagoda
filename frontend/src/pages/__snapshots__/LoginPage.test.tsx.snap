// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-3lrezk"
      >
        <div
          class="MuiBox-root css-105lrk6"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 css-1w8gp40-MuiTypography-root"
          >
            AirOne
          </h1>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-jpb38c-MuiTypography-root"
          />
          <div
            class="MuiBox-root css-4ubqiz"
          />
          <form
            class="MuiBox-root css-ttwlo1"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="username"
                id="username-label"
              >
                Username
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="username"
                  name="username"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-q0jhri-MuiInputBase-root-MuiInput-root"
            >
              <input
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="next"
                name="next"
                type="hidden"
                value=""
              />
            </div>
            <div
              class="MuiBox-root css-10uuh3u"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-5fap85-MuiTypography-root-MuiLink-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ita7xj-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-11fipy7-MuiTypography-root"
                />
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-5fap85-MuiTypography-root-MuiLink-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ita7xj-MuiSvgIcon-root"
                  data-testid="InfoIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-11fipy7-MuiTypography-root"
                >
                  パスワードリセット
                </p>
              </a>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1o2haby-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-3lrezk"
    >
      <div
        class="MuiBox-root css-105lrk6"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1w8gp40-MuiTypography-root"
        >
          AirOne
        </h1>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-jpb38c-MuiTypography-root"
        />
        <div
          class="MuiBox-root css-4ubqiz"
        />
        <form
          class="MuiBox-root css-ttwlo1"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              Username
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-q0jhri-MuiInputBase-root-MuiInput-root"
          >
            <input
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="next"
              name="next"
              type="hidden"
              value=""
            />
          </div>
          <div
            class="MuiBox-root css-10uuh3u"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-5fap85-MuiTypography-root-MuiLink-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ita7xj-MuiSvgIcon-root"
                data-testid="InfoIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-11fipy7-MuiTypography-root"
              />
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-5fap85-MuiTypography-root-MuiLink-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ita7xj-MuiSvgIcon-root"
                data-testid="InfoIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body1 css-11fipy7-MuiTypography-root"
              >
                パスワードリセット
              </p>
            </a>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1o2haby-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
